### PR TITLE
BridgeCrew compliance updates

### PR DESCRIPTION
**Pull request by mclacore**
Compliance updates for Cosmos DB Mongo adding customer-managed key encryption.

New resources added:
* Azure Key Vault
* Azure Key Vault key

Once this gets merged to main, I'll open an identical PR for `azure-cosmosdb-sql`

closes: https://github.com/massdriver-cloud/massdriver/issues/1367

### CODE REVIEW

**Comment by mclacore** [comment]
Potential problem when deleting a deployed bundle with CMK:
```
│ Error: cannot purge key "localdevcosmosmongo0015" because vault "Vault: (Name \"localdevcosmosmongo0015\" / Resource Group \"local-dev-cosmosmongo-0015\")" has purge protection enabled
```
Continuing testing.

**Review by coryodaniel — APPROVED** [review]

File: src/_providers.tf [hunk 1]
```diff
@@ -24,3 +28,9 @@ provider "azurerm" {
   client_secret   = var.azure_service_principal.data.client_secret
   subscription_id = var.azure_service_principal.data.subscription_id
 }
+
+provider "azuread" {
```
Comment (coryodaniel) [inline src/_providers.tf:32]: @chrisghill will need to review I don’t k ow if we have this provider in XO yet.

File: massdriver.yaml [hunk 2]
```diff
@@ -87,6 +73,34 @@ params:
           $md.immutable: true
           type: boolean
           default: false
+      dependencies:
```
Comment (jaketf) [inline massdriver.yaml:76]: this change is not in PR description. Was it meant to be in this PR?

File: src/main.tf [hunk 3]
```diff
@@ -27,25 +17,30 @@ resource "azurerm_resource_group" "main" {
...
 
   enable_automatic_failover       = var.geo_redundancy.automatic_failover
   enable_multiple_write_locations = var.geo_redundancy.multi_region_writes
   mongo_server_version            = var.database.mongo_server_version
 
-  virtual_network_rule {
-    id = azurerm_subnet.main.id
-  }
-
+  # If we wanted to use RBAC instead of access policies for CMK, we would need to set up a two-step deployment for:
```
Comment (jaketf) [inline src/main.tf:34]: is this a TODO or just a lesson you learned while authoring this PR? 
if the former, let's make a GH issue.
if the later, I don't think this needs to be part of the codebase as a reader is just prone to think "idk should we use RBAC instead?".

**Review by jaketf — APPROVED** [review]

**Pull request merged**